utils.py

```diff
@@ -926,4 +929,103 @@ def quote_field_name(f):
+
+    Merges multiple TStructs together and outputs a new TStruct with the union of the fields.
+    Notes:
+    - If a leaf field is found in multiple input TStructs, its type has to be the same
+    - Attributes are merged. If the same attribute is found in different TStructs it should have the same value.
+
+    :param list s: List of Structs to merge
+    :return: Merged Struct
+    :rtype: TStruct
+
+    """
+
+    if len(s) < 2:
+        return s.pop()
+
+    fields = OrderedDict()
+    s_fields = [flatten_struct(x, root='', recursive=False) for x in s]
+
+    while len(s_fields) > 0:
+        s_current = s_fields.pop(0)
+        for name, f in s_current.iteritems():
+            if name not in fields:
+                attributes = f.attributes
+                f_overlap = [x[name] for x in s_fields if name in x]
+
+                for f2 in f_overlap:
+                    if not isinstance(f2.typ, type(f.typ)):
+                        sys.exit("ERROR: Cannot merge structs with type {} and {}".format(f.typ, f2.typ))
+                    for k,v in f2.attributes.iteritems():
+                        if k in attributes:
+                            if v != attributes[k]:
+                                logger.warn("Found different values for attribute {} for field {} while merging structs:{}, {}".format(k,name,attributes[k],v))
+                        else:
+                            attributes[k] = v
+
+                if isinstance(f.typ, TStruct) and f_overlap:
+                    fields[name] = Field(name, merge_TStructs([f.typ] + [f2.typ for f2 in f_overlap]))
+                else:
+                    fields[name] = f
+
+                fields[name].attributes = attributes
+
+    return TStruct.from_fields(fields.values())
+
+
+def replace_vds_variant_schema(vds, new_schema):
+    """
+
+    Replaces the input VDS va with the new schema. Values for all fields present in the old variant schema
+    that have the same type are kept. All other fields are filled with `NA`
+
+    :param VariantDataset vds: input VDS
+    :param TStruct new_schema: new schema
+    :return: VDS with new schema
+    :rtype: VariantDataset
+    """
+
+    def get_schema_expr(struct, root, old_schema_fields):
+
+        field_expr = []
+
+        for f in struct.fields:
+            path = '{}.{}'.format(root, f.name)
+            if not path in old_schema_fields.keys():
+                field_expr.append('{}: NA:{}'.format(f.name, f.typ))
+            elif isinstance(f.typ, TStruct):
+                field_expr.append('{}: {}'.format(f.name, get_schema_expr(f.typ, path, old_schema_fields)))
+            else:
+                field_expr.append('{0}: {1}'.format(f.name, path))
+
+        print '{{{}}}'.format(",".join(field_expr))
+        return '{{{}}}'.format(",".join(field_expr))
+
+    for path, field in flatten_struct(new_schema, root='va').iteritems():
+        if field.attributes:
+            vds = vds.set_va_attributes(path, field.attributes)
```

Comment:
I thought rewriting a schema overwrites the attributes. If that's the case, shouldn't this come after the `annotate_variants_expr`? I could be wrong